validphys2/src/validphys/commondataparser.py

```diff
@@ -40,22 +108,825 @@
     "SIA": ("$z$", "$Q^2 (GeV^2)$", "$y$"),
 }
 
+PROCESS_DESCRIPTION_LABEL = {
+    "EWJ_JRAP": "Jet Rapidity Distribution",
+    "EWK_RAP": "Drell-Yan Rapidity Distribution",
+    "EWJ_RAP": "Jet Rapidity Distribution",
+    "HQP_PTQ": "Heavy Quarks Production Single Quark Transverse Momentum Distribution",
+    "JET": "Jets Rapidity Distribution",
+    "HIG_RAP": "Higgs Rapidity Distribution",
+    "HQP_YQ": "Heavy Quarks Production Single Quark Rapidity Distribution",
+    "EWJ_JPT": "Jet Transverse Momentum Distribution",
+    "DIS": "Deep Inelastic Scattering",
+    "HQP_PTQQ": "Heavy Quarks Production Transverse Momentum Distribution",
+    "EWK_PT": "Drell-Yan Transverse Momentum Distribution",
+    "EWJ_PT": "Jet Transverse Momentum Distribution",
+    "PHT": "Photon Production",
+    "HQP_MQQ": "Heavy Quarks Production Mass Distribution",
+    "EWK_PTRAP": "Drell-Yan Transverse Momentum Distribution",
+    "HQP_YQQ": "Heavy Quarks Production Rapidity Distribution",
+    "INC": "Heavy Quarks Total Cross Section",
+    "EWJ_MLL": "Jet Mass Distribution",
+    "EWK_MLL": "Drell-Yan Mass Distribution",
+    "DIJET": "Dijets Invariant Mass and Rapidity Distribution",
+    "DYP": "Fixed-Target Drell-Yan",
+}
+
+
+def _get_ported_kinlabel(process_type):
+    """Get the kinematic label for ported datasets
+    In principle there is a one to one correspondance between the process label in the kinematic and
+    ``KINLABEL_LATEX``, however, there were some special cases that need to be taken into account
+    """
+    if process_type in KINLABEL_LATEX:
+        return KINLABEL_LATEX[process_type]
+    # special case in which the process in DIS- or DYP-like
+    if process_type[:3] in ("DIS", "DYP"):
+        return _get_ported_kinlabel(process_type[:3])
+    raise KeyError(f"Label {process_type} not recognized in KINLABEL_LATEX")
+
+
+def _get_process_description(process_type):
+    """Get the process description string for a given process type
+    Similarly to kinlabel, some special cases are taken into account.
+    """
+    if process_type in PROCESS_DESCRIPTION_LABEL:
+        return PROCESS_DESCRIPTION_LABEL[process_type]
+    # If not, is this a DYP- or DIS-like dataset?
+    if process_type[:3] in ("DIS", "DYP"):
+        return _get_process_description(process_type[:3])
+    # Remove pieces of "_" until it is found
+    if len(process_type.split("_")) > 1:
+        return _get_process_description(process_type.rsplit("_", 1)[0])
```

Comment:
> do you expect a process type to be DISXX or DYXX in which case it shouldn't be because it should always be DIS_XX or DY_XX.

To be completely honest I don't know. This is what was being done in the old reader so I just copied it, I didn't check whether one of them was DISXX.

Hopefully through #1949 this won't be necessary anymore. For now I'll leave this (and put it in _get_ported_kinlabel, since as you say it might occur as well).

Reply:
At least in the documentation (https://docs.nnpdf.science/data/exp-data-files.html#process-types-and-kinematics) the convention should be that the naming should be constructed in segments - I interpreted this as separated by `_`.

Reply:
Yes, but we also have DIJET for instance so maybe there's some DISHERA or something somewhere?

(I honestly don't know, I first found we  have a few `process_type[:3] == DIS` scattered around the code not that long ago...)